FrontEnd/src/components/SearchPage/Search.jsx

```diff
@@ -10,53 +8,52 @@ import link_to_left from './img/link_to_left.svg';
 import link_to_right from './img/link_to_right.svg';
 import styles from './search_page.module.css';
 import PropTypes from 'prop-types';
+import useSWR from 'swr';
 
 const ITEMS_PER_PAGE = 6;
 
-export function Search({ isAuthorized, userData }) {
+export function Search({ isAuthorized }) {
   Search.propTypes = {
-    isAuthorized: PropTypes.any.isRequired,
-    userData: PropTypes.any.isRequired,
+    isAuthorized: PropTypes.bool,
   };
 
   const [searchResults, setSearchResults] = useState([]);
   const [searchPerformed, setSearchPerformed] = useState(false);
-  const [error, setError] = useState(null);
 
   const location = useLocation();
   const searchParams = new URLSearchParams(location.search);
   const searchTerm = searchParams.get('name');
   const servedAddress = process.env.REACT_APP_BASE_API_URL;
   const searchUrl = 'search';
-  const { mutate } = useSWRConfig();
-
-  const fetcher = (url) => axios.get(url).then((res) => res.data);
+  const authToken = localStorage.getItem('Token');
+  const headers = authToken
+    ? {
+        withCredentials: true,
+        headers: {
+          Authorization: `Token ${authToken}`,
+        },
+      }
+    : {
+        'Content-Type': 'application/json',
+      };
 
-  async function getRequest(url) {
-    const data = await fetcher(url);
-    setSearchResults(data);
-    setSearchPerformed(true);
-    setError(null);
-  }
+  const fetcher = async (url) => {
+    await axios.get(url, headers).then((res) => {
+      setSearchResults(res.data.results);
+      return res.data.results;
+    });
+  };
 
-  const { trigger } = useSWRMutation(
-    `${servedAddress}/api/search/?name=${searchTerm}`,
-    getRequest
+  const { data: companylist, error } = useSWR(
+    `${servedAddress}/api/profiles/?new_members=-completeness,-created_at`,
+    fetcher
   );
 
-  mutate((key) => typeof key === 'string' && key.startsWith('/api/search/'), {
-    revalidate: true,
-  });
-
   useEffect(() => {
     if (searchTerm) {
-      try {
-        trigger();
-      } catch (error) {
-        console.error(error);
-      }
+      setSearchPerformed(true);
     }
-  }, [searchTerm, servedAddress, searchUrl]);
+  }, [searchTerm, servedAddress, searchUrl, authToken, companylist]);
```

Comment:
`authToken` should be wrapped in useMemo and have isAuth as a dependency to work properly here. Ideally to do it within `useProvideAuth`. I think Roman is working on it